include/osquery/events.h

```diff
@@ -341,7 +347,7 @@ class EventPublisher : public EventPublisherCore {
    *
    * @return should the Subscription%'s EventCallback be called for this event.
    */
-  virtual bool shouldFire(const SCRef sc, const ECRef ec) { return true; }
+  virtual bool shouldFire(const SCRef& sc, const ECRef& ec) { return true; }
```

Comment:
This is a systemic oversight and may have caused unknown behavior with how event publishers implemented their shouldFire methods. For inotify, the `fire` method would not have called the publisher-specific method and thus returned true for every event.